src/content/peerconnection/restart-ice/js/main.js

```diff
@@ -261,14 +261,22 @@ function checkStats(pc) {
     var activeCandidatePair = null;
     var remoteCandidate = null;
 
-    // search for the candidate pair
+    // Search for the candidate pair, spec-way first.
     results.forEach(function(report) {
-      if (report.type === 'candidate-pair' && report.state === 'succeeded' &&
-          report.selected || report.type === 'googCandidatePair' &&
-          report.googActiveConnection === 'true') {
-        activeCandidatePair = report;
+      if (report.type === 'transport') {
+        activeCandidatePair = results.get(report.selectedCandidatePairId);
       }
     });
+    // Fallback for Firefox and Chrome legacy stats.
+    if (!activeCandidatePair) {
+      results.forEach(function(report) {
+        if (report.type === 'candidate-pair' && report.state === 'succeeded' &&
+            report.selected || report.type === 'googCandidatePair' &&
```

Comment:
It bothers me to see things broken up this way. Can you add parentheses and linebreaks so that the || appears visually between the && sequences?

Reply:
re-reviewing this I should actually remove the legacy way :-)
Will do here and in constraints

Reply:
Does that mean it'll stop working on version <old>? Better document that in a comment or something...

Reply:
no backward compat contract here. spec-stats have been implemented for way over a year so...